### PR TITLE
Bump vcpkg-robotology-dependencies tag to v0.10.1 and fix qhull linking

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/robotology/robotology-superbuild/issues/1266 .

I am not a big fan of continuining to support vcpkg, but I think that in this case it detected a real issue: we link the available target of the library we depend on for all dependencies without explicitly specifying if we use the static or shared version, but for some reason for qhull (C version) we are linking the static version. The latest version of qhull on vcpkg does not include also the static version (if I am not wrong), so linking explicitly the static will fail.

### CODE REVIEW

**Comment by traversaro** [comment]
> Fix [robotology/robotology-superbuild#1266](https://github.com/robotology/robotology-superbuild/issues/1266) .
> 
> I am not a big fan of continuining to support vcpkg, but I think that in this case it detected a real issue: we link the available target of the library we depend on for all dependencies without explicitly specifying if we use the static or shared version, but for some reason for qhull (C version) we are linking the static version. The latest version of qhull on vcpkg does not include also the static version (if I am not wrong), so linking explicitly the static will fail.

Interestingly, according to https://github.com/ami-iit/bipedal-locomotion-framework/issues/52#issuecomment-646874888 the link to `qhullstatic_r` was added to make the windows build work.

**Comment by traversaro** [comment]
Probably this is also related to https://github.com/ami-iit/bipedal-locomotion-framework/pull/51 . At the moment qhull on Windows was compiled from source, probably now we can switch to use the one from vcpkg. However, we need to also switch to use `vcpkg-robotology-with-gazebo.zip` as qhull is installed there as a transitive dependency of PCL and VTK.

**Comment by traversaro** [comment]
> Now the build fails with:
> 
> ```
> 2022-09-20T12:36:52.0809895Z      Creating library D:/a/bipedal-locomotion-framework/bipedal-locomotion-framework/build/lib/Debug/BipedalLocomotionFrameworkParametersHandlerYarpImplementationd.lib and object D:/a/bipedal-locomotion-framework/bipedal-locomotion-framework/build/lib/Debug/BipedalLocomotionFrameworkParametersHandlerYarpImplementationd.exp
> 2022-09-20T12:36:52.2781772Z   ParametersHandlerYarpImplementation.vcxproj -> D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\build\bin\Debug\BipedalLocomotionFrameworkParametersHandlerYarpImplementationd.dll
> 2022-09-20T12:36:53.0285579Z   Building Custom Rule D:/a/bipedal-locomotion-framework/bipedal-locomotion-framework/src/Perception/CMakeLists.txt
> 2022-09-20T12:36:53.1174342Z   ArucoDetector.cpp
> 2022-09-20T12:36:54.4252842Z D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\src\Perception\src\ArucoDetector.cpp(18,10): fatal error C1083: Cannot open include file: 'opencv2/aruco.hpp': No such file or directory [D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\build\src\Perception\PerceptionFeatures.vcxproj]
> 2022-09-20T12:36:54.5463790Z   Building Custom Rule D:/a/bipedal-locomotion-framework/bipedal-locomotion-framework/src/RobotInterface/YarpImplementation/CMakeLists.txt
> 2022-09-20T12:36:54.6363564Z   YarpCameraBridge.cpp
> 2022-09-20T12:36:56.3502851Z D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\install\deps\include\yarp/dev/FrameGrabberInterfaces.h(11): warning : "<yarp/dev/FrameGrabberInterfaces.h> file is deprecated" [D:\a\bipedal-locomotion-framework\bipedal-locomotion-framework\build\src\RobotInterface\YarpImplementation\PerceptionInterfaceYarpImplementation.vcxproj]
> 2022-09-20T12:36:58.1517481Z   Auto build dll exports
> ```
> 
> Probably the problem is that `contrib` (of which aruco is part) is not anymore a default feature of opencv in vcpkg, so we need to enable it explcitly: [robotology/robotology-superbuild-dependencies-vcpkg#57](https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/pull/57) .

I created https://github.com/robotology/robotology-superbuild-dependencies-vcpkg/releases/tag/v0.10.1 to fix this, and now the PR is working fine and is ready for review. @GiulioRomualdi @S-Dafarra

**Comment by traversaro** [comment]
@GiulioRomualdi @S-Dafarra The CI ended successfully!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
   - cron:  '0 2 * * *'
 
 env:
-  vcpkg_robotology_TAG: v0.8.0
+  vcpkg_robotology_TAG: v0.10.1
```
Comment (GiulioRomualdi) [inline .github/workflows/ci.yml:14]: I would also remove `Qhull_TAG: 2020.2` since it is not used anymore.

**Review by GiulioRomualdi — COMMENTED** [review]
One minor comment

File: .github/workflows/ci.yml (hunk 1 above)
Comment (traversaro) [reply .github/workflows/ci.yml:14]: It is still used in Ubuntu: https://github.com/ami-iit/bipedal-locomotion-framework/blob/traversaro-patch-3/.github/workflows/ci.yml#L460 .

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -130,7 +130,7 @@ jobs:
...
+        key: blf-source-deps-${{ runner.os }}-os-${{ matrix.os }}-build-type-${{ matrix.build_type }}-vcpkg-robotology-${{ env.vcpkg_robotology_TAG }}-ycm-${{ env.YCM_TAG }}-yarp-${{ env.YARP_TAG }}-iDynTree-${{ env.iDynTree_TAG }}-catch2-${{ env.Catch2_TAG }}-casADi-${{ env.CasADi_TAG }}-manif-${{ env.manif_TAG }}-matioCpp-${{ env.matioCpp_TAG }}-LieGroupControllers-${{ env.LieGroupControllers_TAG }}-osqp-${{ env.osqp_TAG }}-osqp-eigen-${{ env.OsqpEigen_TAG }}-tomlplusplus-${{ env.tomlplusplus_TAG }}-unicycle-${{ env.UnicyclePlanner_TAG }}-icub-models-${{ env.icub_models_TAG }}-robometry-${{ env.telemetry_TAG }}
```
Comment (GiulioRomualdi) [inline .github/workflows/ci.yml:133]: Since it is used in ubuntu this shouldn't be modified right?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (GiulioRomualdi) [reply .github/workflows/ci.yml:14]: You're right so why it has been removed from here: https://github.com/ami-iit/bipedal-locomotion-framework/pull/565/files#r978404248

File: .github/workflows/ci.yml (hunk 2 above)
Comment (traversaro) [reply .github/workflows/ci.yml:133]: Good point, I tought this key was Windows only!

Comment (traversaro) [reply .github/workflows/ci.yml]: (I was confused as vcpkg-robotology that is a Windows-only stuff was used there)

Fixed in https://github.com/ami-iit/bipedal-locomotion-framework/pull/565/commits/aafcc6712109f4c120187c4dccbbd98e55db0698 .

**Comment by GiulioRomualdi** [comment]
I rearrange the commits and I will merge it :)

**Pull request merged**